src/UI/SlotArea.cpp

```diff
@@ -1126,8 +1127,137 @@ void cSlotAreaAnvil::UpdateResult(cPlayer & a_Player)
+				&& ((CurrentEnchantment != 0) && (CurrentEnchantment != 1) && (CurrentEnchantment != 3) && (CurrentEnchantment != 4) && (CurrentEnchantment != 7) && (CurrentEnchantment != 34)))
+				/* Chestplate */
+				|| (((Input.m_ItemType == 300) || (Input.m_ItemType == 304) || (Input.m_ItemType == 308) || (Input.m_ItemType == 312) || (Input.m_ItemType == 316))
+				&& ((CurrentEnchantment != 0) && (CurrentEnchantment != 1) && (CurrentEnchantment != 3) && (CurrentEnchantment != 4) && (CurrentEnchantment != 7) && (CurrentEnchantment != 34)))
+				/* Leggings */
+				|| (((Input.m_ItemType == 301) || (Input.m_ItemType == 305) || (Input.m_ItemType == 309) || (Input.m_ItemType == 313) || (Input.m_ItemType == 317))
+				&& ((CurrentEnchantment != 0) && (CurrentEnchantment != 1) && (CurrentEnchantment != 2) && (CurrentEnchantment != 3) && (CurrentEnchantment != 4) && (CurrentEnchantment != 7) && (CurrentEnchantment != 8) && (CurrentEnchantment != 34)))
+				/* Boots */
+				|| (((Input.m_ItemType == 268) || (Input.m_ItemType == 272) || (Input.m_ItemType == 267) || (Input.m_ItemType == 276)|| (Input.m_ItemType == 283))
+				&& ((CurrentEnchantment != 16) && (CurrentEnchantment != 17) && (CurrentEnchantment != 18) && (CurrentEnchantment != 19) && (CurrentEnchantment != 20) && (CurrentEnchantment != 21) && (CurrentEnchantment != 34)))
+				/* Swords */
+				|| (((Input.m_ItemType == 271) || (Input.m_ItemType == 275) || (Input.m_ItemType == 258) || (Input.m_ItemType == 279) || (Input.m_ItemType == 286))
+				&& ((CurrentEnchantment != 16) && (CurrentEnchantment != 17) && (CurrentEnchantment != 18) && (CurrentEnchantment != 32) && (CurrentEnchantment != 33) && (CurrentEnchantment != 34) && (CurrentEnchantment != 35)))
+				/* Axe */
+				|| (((Input.m_ItemType == 269) || (Input.m_ItemType == 273) || (Input.m_ItemType == 256) || (Input.m_ItemType == 277) || (Input.m_ItemType == 284))
+				&& ((CurrentEnchantment != 32) && (CurrentEnchantment != 33) && (CurrentEnchantment != 34) && (CurrentEnchantment != 35)))
+				/* Shovels */
+				|| (((Input.m_ItemType == 290) || (Input.m_ItemType == 291) || (Input.m_ItemType == 292) || (Input.m_ItemType == 293) || (Input.m_ItemType == 294))
+				&& (CurrentEnchantment != 34))
+				/* Hoes */
+				|| ((Input.m_ItemType == 261)
+				&& ((CurrentEnchantment != 48) && (CurrentEnchantment != 49) && (CurrentEnchantment != 50) && (CurrentEnchantment != 51) && (CurrentEnchantment != 34)))
+				/* Bow */
+				|| ((Input.m_ItemType == 346)
+				&& ((CurrentEnchantment != 34) && (CurrentEnchantment != 61) && (CurrentEnchantment != 62)))
+				/* Fishing Rod */
+				|| ((Input.m_ItemType == 398)
+				&& (CurrentEnchantment != 34))
+				/* Carrot on a Stick */
+				|| ((Input.m_ItemType == 359)
+				&& ((CurrentEnchantment != 34) && (CurrentEnchantment != 32) && (CurrentEnchantment != 33)))
+				/* Shears */
+				|| ((Input.m_ItemType == 259)
+				&& (CurrentEnchantment != 34))
+				/* Flint and Steel */
+				)
```

Comment:
This is just TERRIBLE!
Use symbolic names (`E_ITEM_ABC`) instead of the numbers. Format the conditions properly, use indent to indicate the depth of the comparisons.